File: output/GPT4o-Mini/xCodeEval_remove_4000/C++/Java/1703e35d41ca5f57aec51234ba058581.java
import java.util.*;
import java.awt.geom.Point2D;

public class 1703e35d41ca5f57aec51234ba058581xcodeeval_processed_4000.json {
    static final double EPS = 1e-9;
    static final double OO = 1e18;
    static final double PI = Math.acos(-1.0);
    
    static class Point {
        double x, y;

        Point(double x, double y) {
            this.x = x;
            this.y = y;
        }

        Point add(Point p) {
            return new Point(this.x + p.x, this.y + p.y);
        }

        Point sub(Point p) {
            return new Point(this.x - p.x, this.y - p.y);
        }

        Point mul(double scalar) {
            return new Point(this.x * scalar, this.y * scalar);
        }

        double length() {
            return Math.sqrt(x * x + y * y);
        }

        double lengthSqr() {
            return x * x + y * y;
        }

        double angle() {
            return Math.atan2(y, x);
        }

        Point rotate(double angle) {
            return new Point(x * Math.cos(angle) - y * Math.sin(angle), 
                             x * Math.sin(angle) + y * Math.cos(angle));
        }
    }

    static double cross(Point a, Point b) {
        return a.x * b.y - a.y * b.x;
    }

    static double dot(Point a, Point b) {
        return a.x * b.x + a.y * b.y;
    }

    static Point vec(Point a, Point b) {
        return new Point(b.x - a.x, b.y - a.y);
    }

    static Point normalize(Point p) {
        double len = p.length();
        return new Point(p.x / len, p.y / len);
    }

    static double length(Point p) {
        return p.length();
    }

    static double id21(Point a, Point b, Point p) {
        if (dot(vec(a, b), vec(a, p)) < EPS)
            return length(vec(a, p));
        if (dot(vec(b, a), vec(b, p)) < EPS)
            return length(vec(b, p));
        return id13(a, b, p);
    }

    static double id13(Point a, Point b, Point p) {
        return Math.abs(cross(vec(a, b), vec(a, p))) / length(vec(a, b));
    }

    static void id1(Point cen, double r, Point p, Point[] r1, Point[] r2) {
        double lensqr = lengthSqr(vec(cen, p));
        if (Math.abs(lensqr - r * r) < EPS) {
            r1[0] = r2[0] = p;
            return;
        }
        Point cp = vec(cen, p);
        double h = length(cp);
        double a = Math.acos(r / h);
        cp = normalize(cp).mul(r);
        r1[0] = cp.rotate(a).add(cen);
        r2[0] = cp.rotate(-a).add(cen);
    }

    static double arcLength(Point a, Point b, double r) {
        double ang = a.angle() - b.angle();
        if (ang < 0) {
            ang += 2 * PI;
        }
        if (ang > PI) {
            ang = 2 * PI - ang;
        }
        return ang * r;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int Xc = scanner.nextInt();
        int Yc = scanner.nextInt();
        int Vc = scanner.nextInt();
        int Xb = scanner.nextInt();
        int Yb = scanner.nextInt();
        int Vb = scanner.nextInt();
        int r = scanner.nextInt();

        Point planet = new Point(Xc, Yc);
        Point ship = new Point(Xb, Yb);
        double R = length(planet);
        double w = Vc / R;
        double left = 0, right = 1L << 30;

        for (int i = 0; i < 300; i++) {
            double mid = (right + left) / 2;
            Point id23 = planet.rotate(w * mid);
            double distance;

            if (id21(ship, id23, new Point(0, 0)) < r) {
                Point[] a = new Point[1], b = new Point[1], c = new Point[1], d = new Point[1];
                id1(new Point(0, 0), r, ship, a, b);
                id1(new Point(0, 0), r, id23, c, d);
                distance = OO;

                distance = Math.min(distance, length(vec(ship, a[0])) + length(vec(c[0], id23)) + arcLength(a[0], c[0], r));
                distance = Math.min(distance, length(vec(ship, b[0])) + length(vec(c[0], id23)) + arcLength(b[0], c[0], r));
                distance = Math.min(distance, length(vec(ship, a[0])) + length(vec(d[0], id23)) + arcLength(a[0], d[0], r));
                distance = Math.min(distance, length(vec(ship, b[0])) + length(vec(d[0], id23)) + arcLength(b[0], d[0], r));
            } else {
                distance = length(vec(ship, id23));
            }

            if (distance / Vb < mid) {
                right = mid;
            } else {
                left = mid;
            }
        }

        System.out.printf("%.10f\n", left);
    }
}

File: output/GPT4o-Mini/xCodeEval_remove_4000/C++/C/1703e35d41ca5f57aec51234ba058581.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <complex.h>
#include <stdbool.h>

#define EPS 1e-9
#define OO 1e18

typedef struct {
    long double x, y;
} point;

typedef enum {
    IN, OUT, BOUNDRY
} STATE;

long double cross(point a, point b) {
    return a.x * b.y - a.y * b.x;
}

long double dot(point a, point b) {
    return a.x * b.x + a.y * b.y;
}

point vec(point a, point b) {
    return (point){b.x - a.x, b.y - a.y};
}

long double length(point a) {
    return sqrt(a.x * a.x + a.y * a.y);
}

long double lengthSqr(point a) {
    return a.x * a.x + a.y * a.y;
}

point normalize(point a) {
    long double len = length(a);
    return (point){a.x / len, a.y / len};
}

point rotate(point p, long double angle) {
    return (point){p.x * cos(angle) - p.y * sin(angle), p.x * sin(angle) + p.y * cos(angle)};
}

point mid(point a, point b) {
    return (point){(a.x + b.x) / 2, (a.y + b.y) / 2};
}

bool intersect(const point a, const point b, const point p, const point q, point *ret) {
    long double d1 = cross(vec(p, a), vec(b, a));
    long double d2 = cross(vec(q, a), vec(b, a));
    *ret = (point){(d1 * q.x - d2 * p.x) / (d1 - d2), (d1 * q.y - d2 * p.y) / (d1 - d2)};
    if (fabs(d1 - d2) > EPS) return true;
    return false;
}

bool id4(const point a, const point b, const point p) {
    return fabs(cross(vec(a, b), vec(a, p))) < EPS && dot(vec(a, p), vec(a, b)) > -EPS && dot(vec(b, a), vec(b, p)) > -EPS;
}

long double id21(const point a, const point b, const point p) {
    if (dot(vec(a, b), vec(a, p)) < EPS)
        return length(vec(a, p));
    if (dot(vec(b, a), vec(b, p)) < EPS)
        return length(vec(b, p));
    return fabs(cross(vec(a, b), vec(a, p))) / length(vec(a, b));
}

int id1(const point cen, const long double r, const point p, point *r1, point *r2) {
    STATE s = circlePoint(cen, r, p);
    if (s != OUT) {
        *r1 = *r2 = p;
        return s == BOUNDRY;
    }
    point cp = vec(cen, p);
    long double h = length(cp);
    long double a = acos(r / h);
    cp = normalize(cp);
    *r1 = (point){cp.x * r + cen.x, cp.y * r + cen.y};
    *r2 = (point){cp.x * r * cos(-a) + cen.x, cp.y * r * sin(-a) + cen.y};
    return 2;
}

long double arc_length(point a, point b, long double r) {
    long double Ang = atan2(a.y, a.x) - atan2(b.y, b.x);
    if (Ang < 0) {
        Ang += 2 * M_PI;
    }
    if (Ang > M_PI) {
        Ang = 2 * M_PI - Ang;
    }
    return Ang * r;
}

int main() {
    int Xc, Yc, Vc, Xb, Yb, Vb, r;
    scanf("%d %d %d", &Xc, &Yc, &Vc);
    scanf("%d %d %d %d", &Xb, &Yb, &Vb, &r);
    point planet = {Xc, Yc};
    point ship = {Xb, Yb};
    long double R = length(planet);
    long double w = Vc / R;
    long double left = 0, right = 1ll << 30;

    for (int i = 0; i < 300; i++) {
        long double mid = (right + left) / 2;
        point id23 = rotate(planet, w * mid);
        long double distance;

        if (id21(ship, id23, (point){0, 0}) < r) {
            point a, b, c, d;
            id1((point){0, 0}, r, ship, &a, &b);
            id1((point){0, 0}, r, id23, &c, &d);
            distance = 1ll << 30;
            distance = fmin(distance, length(vec(ship, a)) + length(vec(c, id23)) + arc_length(a, c, r));
            distance = fmin(distance, length(vec(ship, b)) + length(vec(c, id23)) + arc_length(b, c, r));
            distance = fmin(distance, length(vec(ship, a)) + length(vec(d, id23)) + arc_length(a, d, r));
            distance = fmin(distance, length(vec(ship, b)) + length(vec(d, id23)) + arc_length(b, d, r));
        } else {
            distance = length(vec(ship, id23));
        }

        if (distance / Vb < mid) {
            right = mid;
        } else {
            left = mid;
        }
    }
    printf("%.10Lf\n", left);
    return 0;
}
